### PR TITLE
Jsdoc support

### PR DESCRIPTION
**Pull request by noraj**
JSDoc support
    
    * update .gitignore for nodejs project + custom rules
    * add JSDoc config file .jsdoc.json
    * update README to explain how to build the documentation
    * update package.json with JSDoc and its minami theme

### CODE REVIEW

**Comment by jfurrow** [comment]
Thanks @noraj1337! This is a great idea! I'll take a look today or tomorrow.

File: README.md [hunk 1]
```diff
@@ -45,15 +51,18 @@ If you have a specific issue or bug, please file a Github issue. If you want to
...
 * Ubuntu users may need to install `nodejs-legacy` (`sudo apt-get install nodejs-legacy`) for dependencies to install successfully. You can read more on [this Stack Overflow post](http://stackoverflow.com/questions/21168141/cannot-install-packages-using-node-package-manager-in-ubuntu).
 
 #### Local Development
+
 1. Run `npm install`.
 2. Run `npm run start:development:server` and `npm run start:development:client` in separate terminal instances.
   * `npm run start:development:server` uses [nodemon](https://github.com/remy/nodemon) to watch for changes to the server-side JavaScript.
   * `npm run start:development:client` watches for changes in the client-side source.
 3. Access the UI through the [WebpackDevServer](https://webpack.js.org/configuration/dev-server/). It expects to proxy requests to the Flood server you have running, defined in `config.js` as `floodServerProxy`.
+4. Build the documentation `./node_modules/.bin/jsdoc -c ./.jsdoc.json`.
```
Comment (jfurrow) [inline README.md]: Would you mind moving this to an npm script and suggesting that here? Something like `npm run build-docs` would be great.

**Comment by noraj1337** [comment]
As I said numerous time I'm not confident with JS. I'm only developing under Ruby, so I'll give you an example with Ruby.

As npm package it exists _gems_ (ruby packages), and you can use _rdoc_ or _yard_ to comment your code like _jsdoc_ does. As the doc is embed in your code (an possibly some other files in your repo) when users install you gem or git clone your repo, they have just to run `yard doc` to build the doc. But also gem are hosted on rubygems.org and rubygems.org automatically build your doc and host it on rubydoc.

Randomly chosen example: 
+ https://rubygems.org/gems/fog-profitbricks
+ http://www.rubydoc.info/gems/fog-profitbricks/4.1.0

But npmjs is not doing that with any documentation tool and flood is not even available as a npm package.

For ruby there are only _rdoc_ and _yard_ that are massively used but for JavaScript it seems there are a lot of documentation tools. JSdoc seemed to me like the most standard.

So users can build the doc locally on their machine. But it is much more great if there is an hosted version of the doc.
There is mainly two solution:
1. generate the doc in the CI and move it to github-pages, [example](https://codeburst.io/generate-docs-and-host-it-with-jsdoc-and-github-pages-113b9dae9acb)
2. Or use a service like [doclets.io](https://doclets.io/)

And yes when the documentation tool will be deployed we must inform contributors about it in contribution guidelines. In your issue #556 you may also want to test that the doc build correctly.

File: README.md (hunk 1 above)
Comment (noraj1337) [reply README.md]: What do you mean? Making a wrapper for the command and tell in the README to launch the wrapper  `npm run build-docs`?

Comment (jfurrow) [reply README.md]: Yes, that's what I mean, sorry that was unclear.

**Comment by jfurrow** [comment]
> generate the doc in the CI and move it to github-pages, example

I like this idea! I think I'll accept your PR now and make another issue for the CI doc generation.

**Comment by noraj1337** [comment]
> What do you think about including a link to http://usejsdoc.org/ (or similar) in the new contributing document?

I created an issue for that.

Include documentation recommendation in the contribution guidelines. See issue #569

**Comment by noraj1337** [comment]
> I like this idea! I think I'll accept your PR (after the npm script alias is implemented).

Is that what you thought about?

**Comment by jfurrow** [comment]
> Is that what you thought about?

Yep, thanks!

**Comment by jfurrow** [comment]
Ugh, I tried to resolve the conflicts using the web UI... it didn't work...

**Comment by noraj1337** [comment]
> Ugh, I tried to resolve the conflicts using the web UI... it didn't work...

@jfurrow I resolved that.

**Comment by noraj** [comment]
@jfurrow Can we merge that ? I really want to write documentation instead of some infos in the wiki.

**Comment by noraj** [comment]
@jfurrow Please can we merge ?

**Comment by jfurrow** [comment]
Sorry @noraj, thanks for your patience and persistence.

**Pull request merged**